validator/accounts/v2/wallet.go

```diff
@@ -102,18 +102,29 @@ func NewWallet(ctx context.Context, cfg *WalletConfig) (*Wallet, error) {
 // OpenWallet instantiates a wallet from a specified path. It checks the
 // type of keymanager associated with the wallet by reading files in the wallet
 // path, if applicable. If a wallet does not exist, returns an appropriate error.
-func OpenWallet(ctx context.Context, cfg *WalletConfig) (*Wallet, error) {
+func OpenWallet(cliCtx *cli.Context, cfg *WalletConfig) (*Wallet, error) {
```

Comment:
This function is now smarter, reducing the need for boilerplate. It is able to determine the kind of keymanager being used as well as the passwords directory if using a direct keymanager